From 4812585772dbaa9bdb5c2aa7df522d615d747b13 Mon Sep 17 00:00:00 2001
From: ricolin <rlin@vexxhost.com>
Date: Tue, 9 Sep 2025 14:24:10 +0800
Subject: [PATCH] fix: disable resizing for admins

By default, admins see all clusters and they are allowed to do all
actions however the resize function will not work so we're displaying
something for admins that they can't use.

This will hide the resize button for clusters that don't match the
project ID of the current user.

Change-Id: If09c509abdd21a5a7b9bc374af52a06404fb0ff8

Change-Id: I5f85af7e3b77840c7efb5873ff03139c714a746d
Signed-off-by: ricolin <rlin@vexxhost.com>
---
 .../clusters/resize/resize.service.js         |  9 ++++---
 .../clusters/resize/resize.service.spec.js    | 27 ++++++++++++++-----
 2 files changed, 26 insertions(+), 10 deletions(-)

diff --git a/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.js b/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.js
index dace892..00099a7 100644
--- a/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.js
+++ b/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.js
@@ -32,6 +32,7 @@
   resizeService.$inject = [
     '$rootScope',
     '$q',
+    'horizon.app.core.openstack-service-api.userSession',
     'horizon.app.core.openstack-service-api.magnum',
     'horizon.framework.util.actions.action-result.service',
     'horizon.framework.util.i18n.gettext',
@@ -43,8 +44,8 @@
   ];
 
   function resizeService(
-    $rootScope, $q, magnum, actionResult, gettext, $qExtensions, modal, toast, spinnerModal,
-    resourceType
+    $rootScope, $q, userSession, magnum, actionResult, gettext, $qExtensions,
+    modal, toast, spinnerModal, resourceType
   ) {
 
     var modalConfig, formModel;
@@ -87,8 +88,8 @@
       return deferred.promise;
     }
 
-    function allowed() {
-      return $qExtensions.booleanAsPromise(true);
+    function allowed(selected) {
+      return userSession.isCurrentProject(selected.project_id);
     }
 
     function constructModalConfig(nodegroups, workerNodesList) {
diff --git a/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.spec.js b/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.spec.js
index 0f64b93..bc49e2c 100644
--- a/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.spec.js
+++ b/magnum_ui/static/dashboard/container-infra/clusters/resize/resize.service.spec.js
@@ -18,10 +18,10 @@
   'use strict';
 
   describe('horizon.dashboard.container-infra.clusters.resize.service', function() {
-
-    var service, $scope, $q, deferred, magnum, spinnerModal, modalConfig;
+    var service, $scope, $q, deferred, magnum, spinnerModal, modalConfig, userSession;
     var selected = {
-      id: 1
+      id: 1,
+      project_id: "f5ed2d21437644adb2669f9ade9c949b"
     };
     var modal = {
       open: function(config) {
@@ -50,6 +50,7 @@
         'horizon.dashboard.container-infra.clusters.resize.service');
       magnum = $injector.get('horizon.app.core.openstack-service-api.magnum');
       spinnerModal = $injector.get('horizon.framework.widgets.modal-wait-spinner.service');
+      userSession = $injector.get('horizon.app.core.openstack-service-api.userSession');
 
       spyOn(spinnerModal, 'showModalSpinner').and.callFake(function() {});
       spyOn(spinnerModal, 'hideModalSpinner').and.callFake(function() {});
@@ -60,9 +61,23 @@
       spyOn(modal, 'open').and.callThrough();
     }));
 
-    it('should check the policy if the user is allowed to update cluster', function() {
-      var allowed = service.allowed();
-      expect(allowed).toBeTruthy();
+    it('should allow user to resize cluster if they are in the same project', async function() {
+      spyOn(userSession, 'get').and.returnValue({project_id: selected.project_id});
+
+      await service.allowed(selected);
+    });
+
+    it('should not allow user to resize cluster if they are in a different project', async function() {
+      spyOn(userSession, 'get').and.returnValue({project_id: 'different_project'});
+
+      try {
+        await service.allowed(selected);
+      } catch (err) {
+        return;
+      }
+
+      throw new Error('User should not be allowed to resize cluster');
+
     });
 
     it('should open the modal, hide the loading spinner and check the form model',
-- 
2.25.1

